src/views/Releases/ListReleases/index.jsx

```diff
@@ -149,16 +227,28 @@ function ListPermissions(props) {
         />
       )}
       <DialogAction
-        title={dialogState.title}
-        body={dialogState.body}
-        open={dialogState.open}
-        error={dialogState.error}
-        confirmText={dialogState.confirmText}
-        onSubmit={handleDialogSubmit}
-        onClose={handleDialogClose}
-        onError={handleDialogError}
-        onComplete={handleDialogClose}
+        title={readOnlyDialogState.title}
+        body={readOnlyDialogState.body}
+        open={readOnlyDialogState.open}
+        error={readOnlyDialogState.error}
+        confirmText={readOnlyDialogState.confirmText}
+        onSubmit={handleReadOnlySubmit}
+        onClose={handleReadOnlyClose}
+        onError={handleReadOnlyError}
+        onComplete={handleReadOnlyClose}
+      />
+      <DialogAction
+        title={deleteDialogState.title}
+        body={deleteDialogState.body}
+        open={deleteDialogState.open}
+        error={deleteDialogState.error}
+        confirmText={deleteDialogState.confirmText}
+        onSubmit={handleDeleteSubmit}
+        onClose={handleDeleteClose}
+        onError={handleDeleteError}
+        onComplete={handleDeleteComplete}
       />
```

Comment:
I think we should be able to handle both actions in a single dialog. We would need to add `onSubmit`, `onClose`, `onError`, `onComplete` to the initial dialog state and set them something like `Function.prototype` or `() => {}`.